estate/models/estate_property.py

```diff
@@ -0,0 +1,137 @@
+                               copy=False, compute="_compute_buyer")
+    salesperson_id = fields.Many2one(
+        "res.users", string="Salesperson", default=lambda self: self.env.uid)
+    tag_ids = fields.Many2many("estate.property.tag", string="Tags")
+    offer_ids = fields.One2many(
+        "estate.property.offer", "property_id", string="Offers")
+    total_area = fields.Integer(
+        compute="_compute_total_area", string="Total Area (sqm)")
+    best_offer = fields.Float(
+        compute="_compute_best_offer")
+
+    _check_expected_price = models.Constraint(
+        'CHECK(expected_price > 0)', 'The expected price must be strictly positive.'
+    )
+    _check_selling_price = models.Constraint(
+        'CHECK(selling_price >= 0)', 'The selling price must be non-negative.'
+    )
+
+    @api.depends('offer_ids.status', 'offer_ids.price')
+    def _compute_selling_price(self):
+        for property in self:
+            accepted_offer = property.offer_ids.filtered(
+                lambda o: o.status == 'accepted')
+            if accepted_offer:
+                property.selling_price = accepted_offer.price
+            else:
+                property.selling_price = 0.0
+
+    @api.depends('offer_ids.status', 'offer_ids.partner_id')
+    def _compute_buyer(self):
+        for property in self:
+            accepted_offer = property.offer_ids.filtered(
+                lambda o: o.status == 'accepted')
+            if accepted_offer:
+                property.buyer_id = accepted_offer.partner_id
+            else:
+                property.buyer_id = None
+
+    @api.depends('living_area', 'garden_area')
+    def _compute_total_area(self):
+        for property in self:
+            property.total_area = property.living_area + property.garden_area
+
+    @api.depends('offer_ids.price')
+    def _compute_best_offer(self):
+        for property in self:
+            if property.offer_ids:
+                property.best_offer = max(property.offer_ids.mapped('price'))
+            else:
+                property.best_offer = 0.0
+
+    @api.constrains('selling_price', 'expected_price')
+    def _check_selling_price_is_acceptable(self):
+        for property in self:
+            if not float_is_zero(property.selling_price, precision_rounding=0.01) and float_compare(
+                property.selling_price,
+                0.9 * property.expected_price,
+                precision_rounding=0.01,
+            ) < 0:
+                raise ValidationError(
+                    "The selling price must be at least 90% of the expected price. You must reduce the expected price if you want to accept this offer."
+                )
+
+    @api.onchange('garden')
+    def _onchange_garden(self):
+        if not self.garden:
+            self.garden_area = 0
+            self.garden_orientation = None
+        else:
+            self.garden_area = 10
+            self.garden_orientation = 'north'
```

Comment:
```suggestion
        if self.garden:
            self.garden_area = 10
            self.garden_orientation = 'north'
        else:
            self.garden_area = 0
            self.garden_orientation = False
```
In odoo we use False for an unset field. Also better to do it this way and not use a not :)